dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ProcessDefinitionService.java

```diff
@@ -295,36 +295,188 @@ private String getResourceIds(ProcessData processData) {
+                    logger.error((String) copyProcessDefinitionResult.get(Constants.MSG));
+                }
+            } catch (Exception e) {
+                failedIdList.add(processDefinitionId);
+            }
+        }
+    }
+
+    /**
+     * check project and auth
+     * @param loginUser
+     * @param projectName
+     * @return
+     */
+    private Map<String, Object> checkProjectAndAuth(User loginUser, String projectName) {
         Project project = projectMapper.queryByName(projectName);
 
+        //check user access for project
         Map<String, Object> checkResult = projectService.checkProjectAndAuth(loginUser, project, projectName);
         Status resultStatus = (Status) checkResult.get(Constants.STATUS);
+
         if (resultStatus != Status.SUCCESS) {
             return checkResult;
         }
+        return null;
+    }
+
+    /**
+     * move process definition
+     * @param processId processId
+     * @param targetProjectName targetProjectName
+     * @return move result code
+     */
+    private Map<String, Object> moveProcessDefinition(Integer processId,
+                                                     String targetProjectName) {
+
+        Map<String, Object> result = new HashMap<>(5);
 
         ProcessDefinition processDefinition = processDefineMapper.selectById(processId);
         if (processDefinition == null) {
             putMsg(result, Status.PROCESS_DEFINE_NOT_EXIST, processId);
             return result;
         } else {
-            return createProcessDefinition(
-                    loginUser,
-                    projectName,
-                    processDefinition.getName() + "_copy_" + System.currentTimeMillis(),
-                    processDefinition.getProcessDefinitionJson(),
-                    processDefinition.getDescription(),
-                    processDefinition.getLocations(),
-                    processDefinition.getConnects());
+            Project targetProject = projectMapper.queryByName(targetProjectName);
+            if(targetProject == null){
+                putMsg(result, Status.PROJECT_NOT_FOUNT, processId);
+                return result;
+            }else{
+                processDefinition.setProjectId(targetProject.getId());
+                processDefinition.setUpdateTime(new Date());
+                if (processDefineMapper.updateById(processDefinition) > 0) {
+                    putMsg(result, Status.SUCCESS);
+                } else {
+                    putMsg(result, Status.UPDATE_PROCESS_DEFINITION_ERROR);
+                }
+                return result;
+            }
+        }
+    }
+
+    /**
+     * check batch operate result
+     * @param result
+     * @param failedIdList
+     */
+    private void checkBatchOperateResult(Map<String, Object> result, List<String> failedIdList) {
+        if (!failedIdList.isEmpty()) {
+            putMsg(result, Status.MOVE_PROCESS_DEFINITION_ERROR, String.join(",", failedIdList));
```

Comment:
It is recommended the error message return project name.

Reply:
already add project name

Reply:
failedIdList -> failedNameList